README.md

```diff
@@ -111,7 +111,7 @@ async fn main() {
     ], Some("book_id")).await.unwrap();
 
     // Query books (note that there is a typo)
-    let query = Query::new("harry pottre");
+    let query = Query::new().with_query("harry pottre").build();
     println!("{:?}", books.search::<Book>(&query).await.unwrap().hits);
```

Comment:
Hey, I maybe have a better idea for this builder, maybe more something like that. What do you think?

```rust
books.search::<Book>().with_query("harry pottre").build().await?;
```

Reply:
That's a great idea! It will be done. But I think we should keep a way to build a `Query` and apply it after.